posts/2018-12-20-Rust-1.31.1.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: "Announcing Rust 1.31.1"
+author: The Rust Release Team
+---
+
+The Rust team is happy to announce a new version of Rust, 1.31.1. Rust is a
+systems programming language focused on safety, speed, and concurrency.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.31.1 is as easy as:
+
+```
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.31.1][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1311-2018-12-20
+
+## What's in 1.31.1 stable
+
+This patch release fixes a build failure on on `powerpc-unknown-netbsd` by
+way of [an update to the `libc`
+crate](https://github.com/rust-lang/rust/pull/56562).
+
+Additionally, the Rust Language Server was updated to fix two critical bugs:
+
+* <https://github.com/rust-lang/rls/issues/1171>
+* <https://github.com/rust-lang/rls/pull/1170>
```

Comment:
I feel like we should have some words here, not just links to issues...

Reply:
Yeah, also because that's the main motivation for the point release. The `libc` update is just a trivial fix we included since we were doing a point release anyway.

Reply:
I don't know a ton about what was fixed and why it was so important, any suggestions?

Reply:
I can try writing something tomorrow, but I'm not too familiar with RLS either. It would be great if @nrc or @Xanewok could write a few lines on the fixes.

Reply:
TL;DR is that some hover requests (for top-most items with doc comments? Would have to double check) caused RLS to enter an infinite loop (causing 100% cpu usage) and the other adjusts rust src path to account for src/ and thus go to definition for libstd stuff works again.

I’ll write some lines in couple of hours.

Reply:
[Hovering over the type with documentation above single-line attributes led to 100% CPU usage:](https://github.com/rust-lang/rls/pull/1170)
```rust
/// Some documentation
#[derive(Debug)] // Multiple, single-line
#[allow(missing_docs)] // atributes
pub struct MyStruct { /* ... */ }
```

[Go to definition fixed for std types](https://github.com/rust-lang/rls/pull/1171):
Before, RLS for `HashMap` tried to open this file
```
~/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/libstd/collections/hash/map.rs
```
and now RLS goes to the correct location (for Rust 1.31, note the extra `src`):
```
~/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/collections/hash/map.rs
```

@alexheretic is my understanding correct for https://github.com/rust-lang/rls/pull/1170?

Reply:
@Xanewok yep, certain hover requests could cause an infinite loop trying to extract documentation. It wasn't as common as every type, only the top type in the source file. I'd just avoid being too specific.